Upgrade dry run mode for vendoring (#1076)
cloudposse/atmos: website/docs/cli/commands/vendor/vendor-pull.mdx

Author: @osterman FYI the comments were raised and addressed in the child PR of 1061, i.e. this one

      version: "latest"
  ```

## Vendoring from SSH

> This is the most common way to access non-public Git repositories from automated systems because it allows access to private repositories without interactive prompts.
> Atmos will use any installed SSH keys automatically. There are two primary ways to specify an SSH source:

Reviewer: ```suggestion
Atmos supports SSH for accessing non-public Git repositories, which is convenient for local development. Atmos will use any installed SSH keys automatically. 

:::tip  
In automated systems like GitHub Actions, we recommend sticking with the `https://` scheme for vendoring. Atmos will automatically inject the `GITHUB_TOKEN`.  
:::

There are two primary ways to specify an SSH source.
```

### SCP-style Sources

Atmos supports traditional SCP-style sources, which use a colon to separate the host from the repository, like this:

```shell
git::git@github.com:cloudposse/terraform-null-label.git?ref={{.Version}}
```

Atmos rewrites this URL to the following format:

```shell
git::ssh://git@github.com/cloudposse/terraform-null-label.git?depth=1&ref={{.Version}}
```

If no username is supplied and the host is `github.com`, Atmos automatically injects the default username `git`.

> ### Explicit SSH

Reviewer: ```suggestion
### Explicit SSH Sources
```

When the SSH scheme (`ssh://`) is explicitly specified, the URL is used as provided. In this scenario, no rewriting occurs because the URL already includes the explicit `ssh://` scheme. For example:

```shell
git::ssh://git@github.com/cloudposse/terraform-null-label.git?ref={{ .Version }}
```

> the URL is used as provided. In this scenario, no rewriting occurs because the URL already includes an explicit `ssh://` schema.

Reviewer: ```suggestion
```

### Important Notes

> - A URL like:

Reviewer: ```suggestion
- The following URL is **invalid** because `go-getter` misinterprets `github.com:` as a URL scheme (like `http:` or `git:`), causing a parsing error:
```

  ```shell
  github.com:cloudposse/terraform-null-label.git?ref={{ .Version }}
  ```

>   is **not valid** because go-getter misinterprets **github.com:** as a schema and will error out.

Reviewer: ```suggestion
```

> - A URL without any scheme, such as:

Reviewer: ```suggestion
- When a URL has no scheme, Atmos defaults to **HTTPS** and injects credentials if available.
```

  ```shell
  github.com/cloudposse/terraform-null-label.git?ref={{ .Version }}
  ```

>   defaults to HTTPS and triggers Atmos to inject credentials if available.  

Reviewer: ```suggestion
```

> ## HTTPS Vendoring

Reviewer: ```suggestion
## Git over HTTPS Vendoring
```

| ATMOS_GITHUB_TOKEN             | Bearer token for GitHub API requests, enabling authentication for private repositories and higher rate limits. |
| GITHUB_TOKEN                   | Used as a fallback if ATMOS_GITHUB_TOKEN is not set.                                                    |

**Default Username for HTTPS:** `"x-access-token"`

### Bitbucket
| Environment Variable            | Purpose                                                                                              |
|---------------------------------|------------------------------------------------------------------------------------------------------|
| ATMOS_BITBUCKET_TOKEN           | Bitbucket app password for API requests; used to avoid rate limits. When both ATMOS_BITBUCKET_TOKEN and BITBUCKET_TOKEN are defined, the former prevails. |
| BITBUCKET_TOKEN                 | Used as a fallback when ATMOS_BITBUCKET_TOKEN is not set.                                           |
| ATMOS_BITBUCKET_USERNAME        | Bitbucket username for authentication. Takes precedence over BITBUCKET_USERNAME.                    |

> | BITBUCKET_USERNAME              | Used as a fallback when ATMOS_BITBUCKET_USERNAME is not set. Bitbucket requires a valid username and does not accept dummy values like `"x-access-token"`. |

Reviewer: Please use definition lists.

```
<dl>
 <dt>term</dt>
 <dd>definition</dd>
</dl>
```

update the others as well.

### Explicit SSH Sources

When the `ssh://` scheme is explicitly specified, the URL is used as provided, and no rewriting occurs.

For example:

```shell
git::ssh://git@github.com/cloudposse/terraform-null-label.git?ref={{ .Version }}
```

### Important Notes

- The following URL is **invalid** because `go-getter` misinterprets `github.com:` as a URL scheme (like `http:` or `git:`), causing a parsing error:
  ```shell
  github.com:cloudposse/terraform-null-label.git?ref={{ .Version }}
  ```

- When a URL has no scheme, Atmos defaults to **HTTPS** and injects credentials if available.
  ```shell
  github.com/cloudposse/terraform-null-label.git?ref={{ .Version }}
  ```

## Git over HTTPS Vendoring

Atmos supports vendoring components using **Git over HTTPS**.

For example:
```
github.com/cloudposse/terraform-null-label.git?ref={{ .Version }}
```
is automatically resolved as:
```
git::https://github.com/cloudposse/terraform-null-label.git?depth=1&ref={{ .Version }}
```

## Authentication & Token Usage for HTTPS

Atmos prioritizes authentication credentials based on predefined environment variables. The priority order for each provider is:

### GitHub

> <dl>
>  <dt>ATMOS_GITHUB_TOKEN</dt>
>  <dd>Bearer token for GitHub API requests, enabling authentication for private repositories and higher rate limits.</dd>
> 
>  <dt>GITHUB_TOKEN</dt>
>  <dd>Used as a fallback if ATMOS_GITHUB_TOKEN is not set.</dd>
> </dl>
> 
> **Default Username for HTTPS:** `"x-access-token"`

Reviewer: ```suggestion
<dl>
 <dt>`ATMOS_GITHUB_TOKEN`</dt>
 <dd>Bearer token for GitHub API requests, enabling authentication for private repositories and higher rate limits.</dd>

 <dt>`GITHUB_TOKEN`</dt>
 <dd>Used as a fallback if `ATMOS_GITHUB_TOKEN` is not set.</dd>
</dl>

**Default Username for HTTPS:** `x-access-token`
```

### Bitbucket

> <dl>
>  <dt>ATMOS_BITBUCKET_TOKEN</dt>
>  <dd>Bitbucket app password for API requests; used to avoid rate limits. When both ATMOS_BITBUCKET_TOKEN and BITBUCKET_TOKEN are defined, the former prevails.</dd>
> 
>  <dt>BITBUCKET_TOKEN</dt>
>  <dd>Used as a fallback when ATMOS_BITBUCKET_TOKEN is not set.</dd>
> 
>  <dt>ATMOS_BITBUCKET_USERNAME</dt>
>  <dd>Bitbucket username for authentication. Takes precedence over BITBUCKET_USERNAME.</dd>
> 
>  <dt>BITBUCKET_USERNAME</dt>
>  <dd>Used as a fallback when ATMOS_BITBUCKET_USERNAME is not set. Bitbucket requires a valid username and does not accept dummy values like `"x-access-token"`.</dd>
> </dl>

Reviewer: ```suggestion
<dl>
 <dt>`ATMOS_BITBUCKET_TOKEN`</dt>
 <dd>Bitbucket app password for API requests; used to avoid rate limits. When both `ATMOS_BITBUCKET_TOKEN` and `BITBUCKET_TOKEN` are defined, the former prevails.</dd>

 <dt>`BITBUCKET_TOKEN`</dt>
 <dd>Used as a fallback when `ATMOS_BITBUCKET_TOKEN` is not set.</dd>

 <dt>`ATMOS_BITBUCKET_USERNAME`</dt>
 <dd>Bitbucket username for authentication. Takes precedence over `BITBUCKET_USERNAME`.</dd>

 <dt>`BITBUCKET_USERNAME`</dt>
 <dd>Used as a fallback when `ATMOS_BITBUCKET_USERNAME` is not set. Bitbucket requires a valid username and does not accept dummy values like `x-access-token`.</dd>
</dl>
```

### GitLab

> <dl>
>  <dt>ATMOS_GITLAB_TOKEN</dt>
>  <dd>Personal Access Token (PAT) for GitLab authentication. Takes precedence over GITLAB_TOKEN.</dd>
> 
>  <dt>GITLAB_TOKEN</dt>
>  <dd>Used as a fallback if ATMOS_GITLAB_TOKEN is not set.</dd>
> </dl>

Reviewer: ```suggestion
<dl>
 <dt>`ATMOS_GITLAB_TOKEN`</dt>
 <dd>Personal Access Token (PAT) for GitLab authentication. Takes precedence over `GITLAB_TOKEN`.</dd>

 <dt>`GITLAB_TOKEN`</dt>
 <dd>Used as a fallback if `ATMOS_GITLAB_TOKEN` is not set.</dd>
</dl>
```
